src/dynamicprompts/samplers/combinatorial.py

```diff
@@ -32,53 +30,30 @@ def _dedupe(arr: list[str]) -> tuple[str, ...]:
 
 
 def _combo_to_prompt(
-    sampler_router: SamplerRouter,
+    sampling_context: SamplingContext,
     combo: list[Command],
 ) -> typing.Iterable[list[str]]:
     if len(combo) == 0:
         yield []
         return
 
     c_1, c_rest = combo[0], combo[1:]
-    gen = sampler_router.generator_from_command(c_1)
+    gen = sampling_context.generator_from_command(c_1)
     if c_1.sampling_method != SamplingMethod.COMBINATORIAL:
```

Comment:
Thinking out loud, I'm not sure whether we will ever find the need to add a new finite sampler, but we may need to check if the sampling method `is_nonfinite` rather then `!= SamplingMethod.ComSamplingMethod.COMBINATORIAL`. A hypothetical finite sampler might be a variable containing a list of values. In that case, it would make sense for it to be treated the same as a combinatorial sampler.